solana/programs/matching-engine/src/fallback/processor/execute_order.rs

```diff
@@ -0,0 +1,706 @@
+    let post_shim_message_account = &accounts[14];
+    let cctp_deposit_for_burn_mint_account = &accounts[15];
+    let cctp_deposit_for_burn_token_messenger_minter_sender_authority_account = &accounts[16];
+    let cctp_deposit_for_burn_message_transmitter_config_account = &accounts[17];
+    let cctp_deposit_for_burn_token_messenger_account = &accounts[18];
+    let cctp_deposit_for_burn_remote_token_messenger_account = &accounts[19];
+    let cctp_deposit_for_burn_token_minter_account = &accounts[20];
+    let cctp_deposit_for_burn_local_token_account = &accounts[21];
+    let cctp_deposit_for_burn_token_messenger_minter_event_authority_account = &accounts[22];
+    let cctp_deposit_for_burn_token_messenger_minter_program_account = &accounts[23];
+    let cctp_deposit_for_burn_message_transmitter_program_account = &accounts[24];
+    let _core_bridge_program_account = &accounts[25];
+    let _core_bridge_config_account = &accounts[26];
+    let _core_bridge_fee_collector_account = &accounts[27];
+    let _post_message_shim_event_authority_account = &accounts[28];
+    let system_program_account = &accounts[29];
+    let token_program_account = &accounts[30];
+
+    // Do checks
+    // ------------------------------------------------------------------------------------------------
+
+    let fast_market_order_data = &fast_market_order_account.data.borrow()[..];
+    let fast_market_order_zero_copy = FastMarketOrderState::try_read(fast_market_order_data)?;
+    // Bind value for compiler (needed for pda seeds)
+    let active_auction_key = active_auction_account.key();
+
+    // Check cctp message is mutable
+    if !cctp_message_account.is_writable {
+        msg!("Cctp message is not writable");
+        return Err(MatchingEngineError::AccountNotWritable.into())
+            .map_err(|e: Error| e.with_account_name("cctp_message"));
+    }
+
+    // Check cctp message seeds
+    let cctp_message_seeds = [
+        common::CCTP_MESSAGE_SEED_PREFIX,
+        active_auction_key.as_ref(),
+    ];
+
+    let (cctp_message_pda, cctp_message_bump) =
+        Pubkey::find_program_address(&cctp_message_seeds, &ID);
+    if cctp_message_pda != cctp_message_account.key() {
+        msg!("Cctp message seeds are invalid");
+        return Err(ErrorCode::ConstraintSeeds.into())
+            .map_err(|e: Error| e.with_pubkeys((cctp_message_pda, cctp_message_account.key())));
+    };
+
+    // Check custodian owner
+    if custodian_account.owner != &ID {
+        msg!(
+            "Custodian owner is invalid: expected {}, got {}",
+            &ID,
+            custodian_account.owner
+        );
+        return Err(ErrorCode::ConstraintOwner.into())
+            .map_err(|e: Error| e.with_account_name("custodian"));
+    };
+
+    // Check custodian deserialises into a checked custodian account
+    let _checked_custodian = Custodian::try_deserialize(&mut &custodian_account.data.borrow()[..])?;
```

Comment:
Isn't there a private method somewhere that performs a custodian check? Can we use that helper method instead? If not, we should create it because there should be several places where we need to check the custodian is the account we expect it to be.